### PR TITLE
feat(profiling): sort samples and import as flamegraph

### PR DESCRIPTION
**Pull request by JonasBa**
Presort the samples based on the type of visualization we want (we sort based on frame pointers because actual frame names are resolved later) and we dont want to resolve the entire frame index twice.

### CODE REVIEW

**Review by eliashussary — APPROVED** [review]
good call on the `ReadOnly` 

all looks good here, i'm excited to see this come together

File: static/app/utils/profiling/profile/importProfile.tsx [hunk 1]
```diff
@@ -96,8 +95,106 @@ export function importProfile(input: ProfileInput, traceID: string): ProfileGrou
...
+// Presort samples so that they are naturally collapsed into a graph.
+// it is important that we do not mutate the original input.
+export function sortProfileSamples<T extends Profiling.ProfileInput>(
+  input: Readonly<T>
+): T {
+  if (isSchema(input)) {
+    const output: T = {...input, profiles: []};
+
+    for (let i = 0; i < input.profiles.length; i++) {
+      // @ts-expect-error we are assigning to a copy
```
Comment (Zylphrex) [inline static/app/utils/profiling/profile/importProfile.tsx:162]: What's the error here?

File: static/app/utils/profiling/profile/sampledProfile.tsx [hunk 2]
```diff
@@ -42,6 +43,7 @@ export class SampledProfile extends Profile {
       });
 
       if (
+        options.type === 'flamechart' &&
```
Comment (Zylphrex) [inline static/app/utils/profiling/profile/sampledProfile.tsx:46]: This feels awfully error prone if we have to check if we're in flamechart/flamegraph mode in all the formats

**Review by Zylphrex — APPROVED** [review]
Really just the comment on the need to check if it's a flamechart vs flamegraph in the various formats to do different things feel awully error prone. Don't have a better suggestion though

File: static/app/utils/profiling/profile/importProfile.tsx (hunk 1 above)
Comment (JonasBa) [reply static/app/utils/profiling/profile/importProfile.tsx:162]: T is inferred as readonly so when we assign to it it screams, but we are creating a copy here anyways

File: static/app/utils/profiling/profile/sampledProfile.tsx (hunk 2 above)
Comment (JonasBa) [reply static/app/utils/profiling/profile/sampledProfile.tsx:46]: Yeah, agree. The other option is to create a different primitive to import these, but that's more work than I'd like to do right now

**Pull request merged**